lib/site.plugin.js

```diff
@@ -1,38 +1,134 @@
 /**
- * Module variables
+ * Module vars
  */
 const root = '/sites';
 
 class SitePlugin {
+
 	/**
 	 * `SitePlugin` constructor.
 	 *
-	 * @param {String} [id] - the plugin ID
+	 * @param {String} [slug] - the plugin slug
 	 * @param {Number|String} sid - site identifier
 	 * @param {WPCOM} wpcom - wpcom instance
 	 * @return {Undefined} undefined
 	 */
-	constructor( id, sid, wpcom ) {
+	constructor( slug, sid, wpcom ) {
 		if ( ! ( this instanceof SitePlugin ) ) {
-			return new SitePlugin( id, sid, wpcom );
+			return new SitePlugin( slug, sid, wpcom );
 		}
 
-		this._id = id;
+		this._slug = encodeURIComponent( slug );
```

Comment:
Since everything in this file is dependent upon the presence of `this._slug` should we add a check here to ensure that a value was passed in properly?